pyccel/ast/cuda.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+CUDA Module
+This module provides a collection of classes and utilities for CUDA programming.
+"""
+from pyccel.ast.core import FunctionCall
+
+__all__ = (
+    'KernelCall',
+)
+
+class KernelCall(FunctionCall):
+    """
+    Represents a kernel function call in the code.
+    
+     The class serves as a representation of a kernel
+     function call within the codebase.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The definition of the function being called.
+
+    args : tuple
```

Comment:
The arguments being passed to the function...
am I mistaken about something  ?

Reply:
The type should be as descriptive as it would need to be in typed code. As written I guess this is equivalent to `tuple[Any,...]`? So if I pass `(1, None, LiteralFloat(4.7), my_func)` to your function everything should work right?

Reply:
Yes, I believe so.

Reply:
I don't :sweat_smile:
It will raise an error here: 
https://github.com/pyccel/pyccel/blob/28f555d3ccd91918e0137ea8d98a9f22a0942797/pyccel/ast/basic.py#L91

Reply:
fixed d6a4291

Reply:
You have changed tuple to list but you still don't specify what types are valid for the elements of the list

Reply:
1:args : List[int]
     list of LiteralInteger passed to the function
2:  args : List of int
     The arguments passed to the function.
if i understand i think the second one is fine ?

Reply:
https://github.com/pyccel/pyccel-cuda/commit/a54eda16b7d2f9a72773de6631eb0933aea20d75

Reply:
1. `args : List[int]`
List of Python integers (e.g. `[1,2,3]`)
2. `args : List[LiteralInteger]`
List of `LiteralInteger` passed to the function
3.  `args : List of int`
Non-standard syntax to specify a list of Python integers (e.g. `[1,2,3]`)

You should use the standard way of specifying types. See https://docs.python.org/3/library/typing.html

Reply:
can u check now please https://github.com/pyccel/pyccel-cuda/commit/adad61a1dbe0f79e7e321632defcff2e708f0d11

Reply:
This is now a complete type :+1: